### PR TITLE
use html5lib>=0.999999999, to fix #403

### LINKED ISSUE #403: Add support for new html5lib versions
Ref:

#336
#344
#345
#402

### PR DESCRIPTION
**Pull request by jedie**
It's a rebased version of #450 and used:
` 'html5lib>=0.999999999'`
instead of:
` 'html5lib==0.999999999'`

### CODE REVIEW

**Comment by jedie** [comment]
ping

**Comment by jedie** [comment]
ping

**Comment by jmaupetit** [comment]
Hi there! Is there any way we can help to finish this WIP and close #403? I'm a bit concerned by the security issue of `html5lib==99999999`.

**Comment by jedie** [comment]
I don't understand why nobody cares.

File: setup.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 INSTALL_REQUIRES = [
     'django-cms>=3.3.0',
-    'html5lib>=0.90,!=0.9999,!=0.99999,<0.99999999',
+    'html5lib>=0.999999999',
```
Comment (toffi9) [inline setup.py]: Also change this version in
```diff
diff --git a/requirements.txt b/requirements.txt
index 6bfcf94..4c35ef1 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,3 +1,3 @@
 django-cms>=3.3
-html5lib>=0.999999
+html5lib>=0.999999999
 Pillow
diff --git a/test_requirements.txt b/test_requirements.txt
index 65078dc..3c0f664 100644
--- a/test_requirements.txt
+++ b/test_requirements.txt
@@ -4,7 +4,7 @@ djangocms-picture>=2.0.6
 djangocms-link>=2.1.2

 Pillow
-html5lib>=0.999999
+html5lib>=0.999999999

 tox>=2.9.1
 coverage>=4.4.2
```

File: djangocms_text_ckeditor/html.py [hunk 2]
```diff
@@ -6,44 +6,41 @@
...
-
-    sanitizer.HTMLSanitizer.acceptable_elements.extend(['cms-plugin'])
+def _filter_kwargs():
+    kwargs = dict(
+        allowed_elements=set(
+            tuple(sanitizer.allowed_elements) + ((namespaces['html'], 'cms-plugin'),)))
 
     if settings.TEXT_HTML_SANITIZE:
-        sanitizer.HTMLSanitizer.acceptable_elements.extend(
+        kwargs['allowed_elements'] = set(
```
Comment (toffi9) [inline djangocms_text_ckeditor/html.py]: When I add `'TEXT_ADDITIONAL_TAGS': ('iframe',),` to test_settings. It won't work. 

Steps to reproduce:

- create page with text plugin
- click `Source` in ckeditor, paste `<iframe src="https://www.divio.com/en/"></iframe>`
- unclick `Source` to go back to normal editing in ckeditor
- click `Save`
- on page should show iframe, not text `<iframe src="https://www.divio.com/en/"></iframe>`

Please check also for ADDITONAL_ATTRIBUTES and ADDITIONAL_PROTOCOLS if it's works properly.

Comment (toffi9) [reply djangocms_text_ckeditor/html.py]: Ideally you can add unit test for that

File: djangocms_text_ckeditor/html.py [hunk 3]
```diff
@@ -6,44 +6,41 @@
...
 from .sanitizer import TextSanitizer
 from .utils import plugin_to_tag
 
 
-def _get_default_parser():
-    opts = {}
-
-    sanitizer.HTMLSanitizer.acceptable_elements.extend(['cms-plugin'])
+def _filter_kwargs():
+    kwargs = dict(
```
Comment (toffi9) [inline djangocms_text_ckeditor/html.py]: could be simply `{}` not `dict()`

File: djangocms_text_ckeditor/html.py [hunk 4]
```diff
@@ -6,44 +6,41 @@
...
-        sanitizer.HTMLSanitizer.allowed_attributes = (
-            sanitizer.HTMLSanitizer.acceptable_attributes +
-            sanitizer.HTMLSanitizer.mathml_attributes +
-            sanitizer.HTMLSanitizer.svg_attributes)
-        sanitizer.HTMLSanitizer.allowed_protocols = (
-            sanitizer.HTMLSanitizer.acceptable_protocols +
-            list(settings.TEXT_ADDITIONAL_PROTOCOLS))
+        kwargs['allowed_protocols'] = set(
+            tuple(sanitizer.allowed_protocols) +
+            tuple(settings.TEXT_ADDITIONAL_PROTOCOLS))
```
Comment (toffi9) [inline djangocms_text_ckeditor/html.py]: ideally last `)` should be in new line to improve readability

File: djangocms_text_ckeditor/html.py [hunk 5]
```diff
@@ -60,9 +57,11 @@ def clean_html(data, full=True, parser=DEFAULT_PARSER):
...
         dom_tree = parser.parseFragment(data)
     walker = treewalkers.getTreeWalker("dom")
-    stream = walker(dom_tree)
-    s = serializer.htmlserializer.HTMLSerializer(omit_optional_tags=False,
-                                                 quote_attr_values=True)
+    kwargs = _filter_kwargs()
+    stream = TextSanitizer(walker(dom_tree), **kwargs)
+    s = serializer.HTMLSerializer(
+        omit_optional_tags=False,
+        quote_attr_values='always')
```
Comment (toffi9) [inline djangocms_text_ckeditor/html.py]: last bracket to new line

File: djangocms_text_ckeditor/html.py (hunk 2 above)
Comment (emekanw) [reply djangocms_text_ckeditor/html.py]: This issue exists now after I updated, rolling back

**Pull request closed without merging**